box_unreal_wrapper.py

```diff
@@ -0,0 +1,83 @@
+from ue5env import UE5EnvWrapper
+import ue5env
+import sys
+from boxnavigator import Action, BoxNavigatorBase
+from math import degrees, radians
+from box import Box, Pt
+
+sys.path.append(
+    "C:/Users/simon/OneDrive/Documents/ArcsLab/ArcLabPrograms/OldenborgTraining/DataCollection"
+)
+from UE5_Data_collector import UE5_data_collection
+
+class NavigatorUnrealWrapper:
+
+    def __init__(
+        self, navigator: BoxNavigatorBase, port: int = 8500, collect_data: bool = False
+    ) -> None:
+        self.experiment_name = "Test"
+        self.ue5 = UE5EnvWrapper(port)
+        self.navigator = navigator
+        self.collect_data = collect_data
+        dataset_path = "C:/Users/simon/OneDrive/Documents/ArcsLab/ArcLabPrograms/UE5Images/small_test_folder"
+        path_to_unreal_project_image = "C:/Users/simon/OneDrive/Documents/Unreal Projects/OldenborgUE/Saved/Screenshots/WindowsEditor/highres.png"
+        self.UE5_data_collector = UE5_data_collection(
+            self.ue5, self.experiment_name, dataset_path, path_to_unreal_project_image
+        )
+        self.syncUnrealPositionToBox()
+        self.syncRotation()
+
+    def syncUnrealPositionToBox(self) -> None:
+        """Move Unreal Agent to match Boxsim Agent X and Y values"""
+        _, _, unrealZ = self.ue5.getCameraLocation(0)
+        self.ue5.setCameraLocation(
+            self.navigator.position.x, self.navigator.position.y, unrealZ
+        )
+
+    def syncBoxPositionToUnreal(self) -> None:
+        """Move Boxsim agent to match Unreal Agent Position"""
+        unrealX, unrealY, _ = self.ue5.getCameraLocation(0)
+        self.navigator.move(Pt(unrealX, unrealY))
+
+    def syncRotation(self) -> None:
+        """Sync Unreal's agent location to box navigator."""
+        # Conversion from Box to unreal location is (180 - boxYaw) = unrealYaw
+        unrealYaw: float = degrees(self.navigator.rotation)
+        self.ue5.setCameraYaw(unrealYaw, 0)
+
+    def take_action(self) -> None:
+        """Performs the same action the box navigator performed but on the Unreal Agent.
+
+        Args:
+            action (Action): The action to perform.
+            navigator (BoxNavigatorBase): The navigator performing the action.
+
+        Raises:
+            RuntimeError: If the action is not defined as Forward, Back, ROTATE_LEFT, or ROTATE_RIGHT.
+        """
+        action_taken, correct_action = self.navigator.take_action()
+        if self.collect_data:
+            self.UE5_data_collector.collectData(correct_action)
+
+        if action_taken == Action.FORWARD:
+            self.ue5.forward(self.navigator.translation_increment)
+        elif action_taken == Action.BACKWARD:
+            self.ue5.back(self.navigator.translation_increment)
+        elif action_taken == Action.ROTATE_LEFT:
+            self.syncRotation()
+        elif action_taken == Action.ROTATE_RIGHT:
+            self.syncRotation()
```

Comment:
Why are some `self.ue5.forward` and some `self.syncRotation`. Can these be made more consistent?